Cpp/odin-views/permission.cpp

```diff
@@ -27,48 +101,48 @@ namespace {
                 const fostlib::string &path,
                 fostlib::http::server::request &req,
                 const fostlib::host &host) const {
-            const auto &permission = config["permission"];
-            bool granted = false;
-            if (odin::c_jwt_trust.value() && req.headers().exists("__jwt")) {
-                const auto &perm_header = req.headers()["__jwt"].subvalue(
-                        odin::c_jwt_permissions_claim.value());
-                if (perm_header) {
-                    auto perms = fostlib::json::parse(perm_header.value());
-                    for (const auto &p : perms) {
-                        if (p == permission) {
-                            granted = true;
-                            break;
+            return check_permission(config, config, path, req, host);
+        }
+    } c_permission;
+
+
+    const class permission_method : public fostlib::urlhandler::view {
+      public:
+        static const fostlib::jcursor userloc;
+
+        permission_method() : view("odin.permission.method") {}
+
+        std::pair<boost::shared_ptr<fostlib::mime>, int> operator()(
+                const fostlib::json &config,
+                const fostlib::string &path,
+                fostlib::http::server::request &req,
+                const fostlib::host &host) const {
+
+            if (req.method() == "HEAD" and config.has_key("GET")) {
+                // HEAD method always use the same permission as GET
+                return check_permission(config["GET"], config, path, req, host);
+            } else if (config.has_key(req.method())) {
```

Comment:
Should this check not be before the one above? That way it is possible to have a specific permission for `HEAD` separate from `GET`. Or maybe we don't want that?

Reply:
From https://tools.ietf.org/html/rfc2616#section-9.4 
```
The HEAD method is identical to GET except that the server MUST NOT
   return a message-body in the response.
```
So I don't think HEAD should have specific permission separate from GET

Reply:
Yeah, I think that's fair enough. I'll merge it once the CI finishes.